contrib/endpoints/src/api_manager/method_impl.cc

```diff
@@ -51,17 +50,17 @@ void MethodInfoImpl::addAudiencesForIssuer(const string &issuer,
   if (iss.empty()) {
     return;
   }
+
   set<string> &audiences = issuer_audiences_map_[iss];
-  stringstream ss(audiences_list);
-  string audience;
-  // Audience list is comma-delimited.
-  while (getline(ss, audience, ',')) {
-    if (!audience.empty()) {  // Only adds non-empty audience.
-      std::string aud = utils::GetUrlContent(audience);
-      if (!aud.empty()) {
-        audiences.insert(aud);
-      }
+
+  // Audience list is comma-delimited with possible white spaces.
+  char *tokens = strtok(const_cast<char *>(audiences_list.c_str()), " ,");
```

Comment:
I think we have to fix the code in fetch_metadata.cc. This is indeed bad behavior. Thank you for pointing out and insisting on the change. Much appreciated :-)